debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java

```diff
@@ -1613,6 +1616,39 @@ public void stopInTheMiddleOfTxAndResume() throws Exception {
         }
     }
 
+    @Test
+    @FixFor("DBZ-1730")
+    public void shouldStartConsumingFromSlotLocation() throws Exception {
+        Testing.Print.enable();
+
+        startConnector(config -> config
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, false)
+                .with(EmbeddedEngine.OFFSET_STORAGE, MemoryOffsetBackingStore.class), true);
+        waitForStreamingToStart();
+
+        consumer = testConsumer(1);
+        executeAndWait("INSERT INTO test_table (text) VALUES ('insert2')");
+        consumer.remove();
+
+        stopConnector();
+        TestHelper.execute(
+                "INSERT INTO test_table (text) VALUES ('insert3');",
+                "INSERT INTO test_table (text) VALUES ('insert4')");
+        startConnector(config -> config
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, true)
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, PostgresConnectorConfig.SnapshotMode.NEVER)
+                .with(EmbeddedEngine.OFFSET_STORAGE, MemoryOffsetBackingStore.class), false);
+
+        consumer.expects(2);
+        consumer.await(TestHelper.waitTimeForRecords() * 5, TimeUnit.SECONDS);
+
+        // We cannot guarantee the flush timing so it is possible that insert2 record will be redelivered
+        Assertions.assertThat(((Struct) consumer.remove().value()).getStruct("after").getString("text")).isIn(Collect.unmodifiableSet("insert2", "insert3"));
```

Comment:
In fact I think you will always get one duplicate record re-emitted due to this logic, which disables skipping of the first flushed record:
```
        boolean skipFirstFlushRecord = true;
        initConnection();

        connect();
        if (offset == null || offset <= 0) {
            offset = defaultStartingPos;
            skipFirstFlushRecord = false;
        }
```

Reply:
@jpechane, any thoughts on this one?

Reply:
@gunnarmorling The problem is that this is important in case of snapthot -> streaming transition. I'd prefer to keep it as is and log a follow-up issue if we want to get it in 1.1.0.CR1. Otherwise we can return back to it later.

Reply:
Re-processing the `confirmed_flush_lsn` is not a problem for me at all.
I was just suggesting that your test could possibly be more explicit. Maybe in the future you could make it available as a config option, but it's not important for me now.

Reply:
@cjbooms - I just improved the test description. In fact we cannot do anything about this as the offset loss is the same as doing snapshot never mode and cannot be separated without intorducing a new snapshot mode or config option.
As this is a real corner case I'd keep it as is.